icsp/deploymentjobs.go

```diff
@@ -134,7 +135,7 @@ func (c *ICSPClient) ApplyDeploymentJobs(buildplans []string, bpdata *OSDPersona
 	var servers []Server
 	// lookup buildplan by name
 	for _, buildPlan := range buildplans {
-		bp, err := c.GetBuildPlanByName(buildPlan)
+		bp, err := c.GetBuildPlanByUri(utils.NewNstring(buildPlan))
```

Comment:
would it be possible to detect when we have a URI vs just a name, then select GetBuildPlanByName when we can see that the buildPlan is not a URI?  this way we don't break the docker-machine plugin on this change.